tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -1319,9 +1319,18 @@ uint32_t SPIRVEmitter::processCall(const CallExpr *callExpr) {
 
     // Evaluate parameters
     for (uint32_t i = 0; i < numParams; ++i) {
-      const auto *arg = callExpr->getArg(i);
+      auto *arg = callExpr->getArg(i);
       const auto *param = callee->getParamDecl(i);
 
+      // In some cases where pass-by-reference should be done (such as an
+      // 'inout' struct), the AST does not use a reference type in the function
+      // signature. The LValueToRValue implicit cast should therefore be ignored
+      // in these cases.
+      if (canActAsOutParmVar(param) && !param->getType()->isReferenceType()) {
+        assert(isa<ImplicitCastExpr>(arg));
```

Comment:
Can actually turn these two lines
```c++
assert(isa<ImplicitCastExpr>(arg));
arg = (dyn_cast<ImplicitCastExpr>(arg))->getSubExpr();
```
into
```c++
arg = cast<ImplicitCastExpr>(arg)->getSubExpr();
```

`cast<T>(v)` will assert internally `v` is of `T` type.